### PR TITLE
Remove hard-coding of RAPIDS version where possible

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
* Add `VERSION` file
* Read `VERSION` file from CMake
* Read `VERSION` file from Python
* Read `kvikio.__version__` from docs build
* Read `VERSION` file from shell scripts
* Use variable substitution in Doxyfile
* Remove unused `ci/checks/style.sh`
* Update `ci/release/update-version.sh`

Issue: https://github.com/rapidsai/build-planning/issues/15

### CODE REVIEW

**Review by robertmaynard — APPROVED** [review]
CMake changes LGTM

**Review by bdice — APPROVED** [review]

**Review by AyodeAwe — APPROVED** [review]

File: ci/build_docs.sh [hunk 1]
```diff
@@ -25,7 +25,9 @@ rapids-mamba-retry install \
   --channel "${PYTHON_CHANNEL}" \
   kvikio libkvikio
 
-export RAPIDS_VERSION_NUMBER="24.04"
+export RAPIDS_VERSION="$(rapids-version)"
+export RAPIDS_VERSION_MAJOR_MINOR="$(rapids-version-major-minor)"
+export RAPIDS_VERSION_NUMBER="$RAPIDS_VERSION_MAJOR_MINOR"
```
Comment (vuule) [inline ci/build_docs.sh:30]: is this one still used anywhere?

Comment (KyleFromNVIDIA) [reply ci/build_docs.sh:30]: Yes, it's used by `rapids-upload-docs`.

**Review by vuule — APPROVED** [review]

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

File: ci/build_docs.sh (hunk 1 above)
Comment (jakirkham) [reply ci/build_docs.sh:30]: Have been confused by the same thing before

Maybe [this reference]( https://github.com/rapidsai/gha-tools/blob/a75d2b5dd1b01129490b478e38f18eb6e35170ac/tools/rapids-upload-docs#L15-L17 ) helps

Honestly think we should consider moving away from requiring environment variables in scripts and switch to adding these as script parameters. It is much easier to understand how the scripts are used then. That's a separate discussion though

**Pull request merged**